docs/src/developer/devel-setup.md

```diff
@@ -26,13 +26,23 @@ folder of the project with:
 
 `docker-compose -f docker/docker-compose.dev.yml up`
 
-To run migrations:
+During the first time running the app, the API will fail because it needs to
+access the database tables that still don't exist, so run the migrations
+to create the tables with:
 
-`docker exec -ti phpreport-db alembic upgrade head`
+`docker exec -ti phpreport-api alembic upgrade head`
 
-To create migrations:
+Kill the containers and run them again, so the API can connect to the database
+correctly.
 
-`docker exec -ti phpreport-db alembic revision --autogenerate -m "Migrations description"`
+[Optional] When running the project for the first time, you can add some
```

Comment:
I'm not sure about marking this as optional. It feels to me that some of the data is a little less optional. The user groups are probably something we'd want to have folks add. And at least one user record is probably also recommended. The data from about line 52 on in the initialData.sql file seem like the most optional of the bunch.

In the migration for adding the task types table, you nicely added an insert statement for some default task types. I wonder if it would be worth updating the initial migration to add the bare metal data needed. Then modify the initialData.sql to include only the optional data (and perhaps rename to optionalInitialData.sql). What do you think?

Reply:
> can we actually get rid of the test.db.Dockerfile?

Not yet, it's still being used for the automated tests: https://github.com/Igalia/phpreport/blob/main/.github/workflows/runtests.yml#L17

> I'm not sure about marking this as optional. It feels to me that some of the data is a little less optional. The user groups are probably something we'd want to have folks add. And at least one user record is probably also recommended. The data from about line 52 on in the initialData.sql file seem like the most optional of the bunch.

> In the migration for adding the task types table, you nicely added an insert statement for some default task types. I wonder if it would be worth updating the initial migration to add the bare metal data needed. Then modify the initialData.sql to include only the optional data (and perhaps rename to optionalInitialData.sql). What do you think?

I'm not sure about adding them to the migrations because even though they are needed for the app to work they should be configured per environment, I'm thinking on the case of setting up a production env for instance. Maybe what we want as a next improvement is a script to insert seed data for dev and production environments? I think only the user groups are more mandatory for any environment. I'm a bit concerned to add hard coded users for any environment. In the case of the task types I've added there because they were hardcoded in the front end, but maybe when we have a seed script it makes sense to also remove them from there?

Reply:
I like the idea of seed data based on environments as a next improvement and moving the Task Type data into one of those.